### PR TITLE
Add DebuggerDisplay to Meter and Instruments

### LINKED ISSUE #87758: Add DebuggerDisplay to System.Diagnostics.Metrics's Meter and Instruments
Meter = Meter.Name (maybe also a count of instruments?)
Instruments = Instrument.Name (maybe also its meter name? And Enabled?). DebuggerDisplay attribute is inherited so the attribute can be added once to the base class.

`DebuggerDisplayAttribute` docs: https://learn.microsoft.com/en-us/visualstudio/debugger/using-the-debuggerdisplay-attribute?view=vs-2022

It could be tagged as help wanted.

### PR DESCRIPTION
**Pull request by pedrobsaila**
Fixes #87758

### CODE REVIEW

**Comment by tommcdon** [comment]
@tarekgh please take a look when you get a moment

**Comment by tarekgh** [comment]
CC @JamesNK as always have a good feedback on the debugger displays :-)

File: src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/Instrument.common.cs [hunk 1]
```diff
@@ -11,6 +11,7 @@ namespace System.Diagnostics.Metrics
     /// <remarks>
     /// This class supports only the following generic parameter types: <see cref="byte" />, <see cref="short" />, <see cref="int" />, <see cref="long" />, <see cref="float" />, <see cref="double" />, and <see cref="decimal" />
     /// </remarks>
+    [DebuggerDisplay("Name = {Name}")]
```
[the remark this answers is not in the document]
Comment (JamesNK) [reply src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/Instrument.common.cs:14]:
```suggestion
    [DebuggerDisplay("Name = {Name}, Meter = {Meter.Name}")]
```
Double-check this works in the debugger.

File: src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/Meter.cs [hunk 2]
```diff
@@ -10,6 +10,7 @@ namespace System.Diagnostics.Metrics
     /// <summary>
     /// Meter is the class responsible for creating and tracking the Instruments.
     /// </summary>
+    [DebuggerDisplay("Name = {Name}, InstrumentCount = {_instruments.Count}")]
```
Comment (JamesNK) [inline src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/Meter.cs:13]:
```suggestion
    [DebuggerDisplay("Name = {Name}, Instruments = {_instruments.Count}")]
```

**Review by JamesNK — APPROVED** [review]
My suggestions are minor improvements, not must haves.

**Comment by tarekgh** [comment]
@pedrobsaila could you please address the feedback and send image how the meter and instrument are displayed under the debugger with your change? Thanks!

File: src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/Instrument.common.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (pedrobsaila) [reply src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/Instrument.common.cs]: .\dotnet.cmd build /t:Test C:\OSS\runtime\src\libraries\System.Diagnostics.DiagnosticSource\tests /p:XUnitOptions="-method System.Diagnostics.Metrics.Tests.MetricsTests.InstrumentCreationTest" displays the following when hovering over `counter` variable :
image

**Review by tarekgh — APPROVED** [review]

**Pull request merged**